From 7b9a08940b6e69c3912a6e3d3e3fc2c13056e889 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mat=C3=ADas=20Hern=C3=A1ndez?= <matiashe@google.com>
Date: Thu, 3 Jul 2025 17:35:17 +0200
Subject: [PATCH] Delete updateNotificationChannelGroupFromPrivilegedListener

This is an unpublished (AIDL only) method and was never used from NLS (or anywhere else).

Bug: 426205822
Test: build
Flag: EXEMPT Code deletion
Change-Id: I65438b1af637f3ba9b75a85f780bdb7867cc4ba8
Merged-In: I65438b1af637f3ba9b75a85f780bdb7867cc4ba8
(cherry picked from commit 6077349a00fdac577592284eadddd18b73c0262e)
---
 core/java/android/app/INotificationManager.aidl       |  1 -
 .../notification/NotificationManagerService.java      | 11 -----------
 2 files changed, 12 deletions(-)

diff --git a/core/java/android/app/INotificationManager.aidl b/core/java/android/app/INotificationManager.aidl
index e2d70f116b01..4a51978a4cc9 100644
--- a/core/java/android/app/INotificationManager.aidl
+++ b/core/java/android/app/INotificationManager.aidl
@@ -165,7 +165,6 @@ interface INotificationManager
     void setOnNotificationPostedTrimFromListener(in INotificationListener token, int trim);
     void setInterruptionFilter(String pkg, int interruptionFilter);
 
-    void updateNotificationChannelGroupFromPrivilegedListener(in INotificationListener token, String pkg, in UserHandle user, in NotificationChannelGroup group);
     void updateNotificationChannelFromPrivilegedListener(in INotificationListener token, String pkg, in UserHandle user, in NotificationChannel channel);
     ParceledListSlice getNotificationChannelsFromPrivilegedListener(in INotificationListener token, String pkg, in UserHandle user);
     ParceledListSlice getNotificationChannelGroupsFromPrivilegedListener(in INotificationListener token, String pkg, in UserHandle user);
diff --git a/services/core/java/com/android/server/notification/NotificationManagerService.java b/services/core/java/com/android/server/notification/NotificationManagerService.java
index af7fc92d2eab..b36e64df9d0b 100644
--- a/services/core/java/com/android/server/notification/NotificationManagerService.java
+++ b/services/core/java/com/android/server/notification/NotificationManagerService.java
@@ -5754,17 +5754,6 @@ public class NotificationManagerService extends SystemService {
             }
         }
 
-        @Override
-        public void updateNotificationChannelGroupFromPrivilegedListener(
-                INotificationListener token, String pkg, UserHandle user,
-                NotificationChannelGroup group) throws RemoteException {
-            Objects.requireNonNull(user);
-            verifyPrivilegedListener(token, user, false);
-            createNotificationChannelGroup(
-                    pkg, getUidForPackageAndUser(pkg, user), group, false, true);
-            handleSavePolicyFile();
-        }
-
         @Override
         public void updateNotificationChannelFromPrivilegedListener(INotificationListener token,
                 String pkg, UserHandle user, NotificationChannel channel) throws RemoteException {
-- 
2.52.0.rc1.455.g30608eb744-goog

